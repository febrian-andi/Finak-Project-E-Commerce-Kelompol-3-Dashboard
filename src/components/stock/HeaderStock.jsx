import React from 'react';
import { Link } from 'react-router-dom';
import PropsTypes from 'prop-types';

const HeaderStock = ({ title, breadcrumbs, onAddStock }) => {
  return (
    <div className='flex justify-between items-center mb-6'>
      <div>
        <h1 className='text-2xl font-semibold mb-1'>{title}</h1>
        <div className='flex items-center text-sm space-x-2'>
          {breadcrumbs.map((breadcrumb, index) => (
            <React.Fragment key={index}>
              <span className='text-red-500'>{breadcrumb}</span>
              {index < breadcrumbs.length - 1 && <span className='text-red-500'>{'>'}</span>}
            </React.Fragment>
          ))}
        </div>
      </div>
      <Link className='bg-red-500 hover:bg-red-600 text-white px-4 py-2 rounded' onClick={onAddStock} to='/stock/add'>
        Add New Stock
      </Link>
    </div>
  );
};

HeaderStock.propTypes = {
  title: PropsTypes.string,
  breadcrumbs: PropsTypes.array,
  onAddStock: PropsTypes.func,
};


export default HeaderStock;
