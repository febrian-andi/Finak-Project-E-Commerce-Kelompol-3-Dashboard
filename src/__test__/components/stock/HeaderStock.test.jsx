import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import HeaderStock from '../../../components/stock/HeaderStock';

describe('HeaderStock Component', () => {
  const defaultProps = {
    title: 'Stock Management',
    breadcrumbs: ['Dashboard', 'Stock'],
    onAddStock: jest.fn(),
  };

  const renderWithRouter = (ui) => {
    return render(<BrowserRouter>{ui}</BrowserRouter>);
  };

  it('renders the title correctly', () => {
    renderWithRouter(<HeaderStock {...defaultProps} />);
    expect(screen.getByText('Stock Management')).toBeInTheDocument();
  });

  it('renders breadcrumbs correctly', () => {
    renderWithRouter(<HeaderStock {...defaultProps} />);

    // Check breadcrumb items
    expect(screen.getByText('Dashboard')).toBeInTheDocument();
    expect(screen.getByText('Stock')).toBeInTheDocument();

    // Check the separator
    const separators = screen.getAllByText('>');
    expect(separators).toHaveLength(1);
  });

  it('renders the Add New Stock button with correct text and link', () => {
    renderWithRouter(<HeaderStock {...defaultProps} />);

    const button = screen.getByRole('link', { name: /add new stock/i });
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('href', '/stock/add');
  });

  it('calls onAddStock when the Add New Stock button is clicked', () => {
    renderWithRouter(<HeaderStock {...defaultProps} />);

    const button = screen.getByRole('link', { name: /add new stock/i });
    fireEvent.click(button);

    expect(defaultProps.onAddStock).toHaveBeenCalled();
  });

  it('handles empty breadcrumbs gracefully', () => {
    renderWithRouter(<HeaderStock {...defaultProps} breadcrumbs={[]} />);

    // Breadcrumb section should not render separators or items
    expect(screen.queryByText('>')).not.toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = renderWithRouter(<HeaderStock {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
