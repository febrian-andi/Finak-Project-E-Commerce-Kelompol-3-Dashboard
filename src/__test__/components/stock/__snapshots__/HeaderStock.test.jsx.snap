// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderStock Component matches snapshot 1`] = `
<div>
  <div
    class="flex justify-between items-center mb-6"
  >
    <div>
      <h1
        class="text-2xl font-semibold mb-1"
      >
        Stock Management
      </h1>
      <div
        class="flex items-center text-sm space-x-2"
      >
        <span
          class="text-red-500"
        >
          Dashboard
        </span>
        <span
          class="text-red-500"
        >
          &gt;
        </span>
        <span
          class="text-red-500"
        >
          Stock
        </span>
      </div>
    </div>
    <a
      class="bg-red-500 hover:bg-red-600 text-white px-4 py-2 rounded"
      data-discover="true"
      href="/stock/add"
    >
      Add New Stock
    </a>
  </div>
</div>
`;
